src/server/session.ts

```diff
@@ -21,6 +21,21 @@ module ts.server {
         }  
         return spaceCache[n];
     }
+    
+    export function generateIndentString(n: number, editorOptions: EditorOptions): string {

```

Comment:
we already have code for that, check [getIndentationString](https://github.com/Microsoft/TypeScript/blob/master/src/services/formatting/formatting.ts#L1065)